internal/stage_common.go

```diff
@@ -18,3 +23,39 @@ func randSeq(n int) string {
 	}
 	return string(b)
 }
+
+func sendRequest(client *http.Client, req *http.Request, logger *testerutils.Logger) (*http.Response, error) {
+	reqDump, err := httputil.DumpRequestOut(req, true)
+	if err != nil {
+		return nil, err
+	}
+	logger.Debugf("Sending Request:\n%s", string(reqDump))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		logFriendlyError(logger, err)
+		return nil, fmt.Errorf("Failed to connect to server, err: '%v'", err)
+	}
+	respDump, err := httputil.DumpResponse(resp, true)
+	if err != nil {
+		return nil, err
+	}
+	logger.Debugf("Received Response:\n%s", string(respDump))
```

Comment:
@sherubthakur as an example - this is good to only keep in debug logs (like you have already). We already have error messages that contain assertion failures, so we don't need an info log for this.